IOIOLibCore/src/main/java/ioio/lib/impl/Board.java

```diff
@@ -36,7 +36,7 @@
 
 	public final Hardware hardware;
 
-	private Board(Hardware hw) {
+	Board(Hardware hw) {
```

Comment:
I am impressed by all the details you've identified, and am learning quite a few things about the preferred Java syntax, and implicit or default modifiers in Java after reviewing this! The only one in this commit that doesn't make sense to me is this one. Why would you make something less encapsulated like this? Was this breaking something?

Reply:
It was pointless, because it's already package protected

Reply:
With Kotlin you'll get one more protection level `internal`

Reply:
It was my understanding that explicitly "private" is one level deeper encapsulation than "package protected" or "default" and the goal is always to keep things as deeply encapsulated as possible. With that in mind, your answer was still was not making sense to me, so I dug a bit deeper. 

Looks like it is not package protected, but "private" by default. In the words of the [Language Spec](https://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.8.3), "If no access modifier is specified for the constructor of an enum type, the constructor is private". Found this in the comments of [this](https://stackoverflow.com/questions/7747948/why-can-a-enum-have-a-package-private-constructor) StackOverflow question. 

After opening this in Android Studio I see that the linter calls this out, but I'll go on assuming you found this without a linter and are just THAT good. Who knows...maybe you actually did! :D Anyways, happy to be learning all these details that I would have otherwise been oblivious to.